docs/badges/details.rst

```diff
@@ -0,0 +1,174 @@
+Implementation Details
+======================
+
+Badges feature is implemented as a separate application within the Credentials Open edX service:
+
+.. code-block::
+
+        credentials/apps/badges/
+
+Events (Event Bus)
+------------------
+
+        Badges feature always relies on events (public signals) with user data.
+
+New events
+~~~~~~~~~~
+
+Badges feature extends the set of already `published events`_ with its own set of new public events:
+
+- `BADGE_AWARDED` (Credentials)
+- `BADGE_REVOKED` (Credentials)
+
+In addition to already present `public events`_ Badges also rely on:
+
+- `COURSE_GRADE_NOW_PASSED` (LMS)
+- `COURSE_GRADE_NOW_FAILED` (LMS)
+
+
+Data models
```

Comment:
Please check this ERD, it's better to have a visual representation of the data model

https://dbdiagram.io/d/Open-edX-Credly-Badges-645cd5efdca9fb07c4e805a4


ERD for the Credentials service can be found there - https://lucid.app/lucidchart/ac49c90e-9718-4f37-9135-f3c0c9a62017/edit?invitationId=inv_8f93357e-0a72-440f-a91a-19f94e78d2a5&page=pt-o3H4-vyK.#